pkgs/development/misc/resholve/deps.nix

```diff
@@ -0,0 +1,150 @@
+    };
+    nativeBuildInputs = [ autoreconfHook ];
+    preCheck = ''
+      patchShebangs run_tests.sh
+    '';
+  };
+
+  py-yajl = python27.pkgs.buildPythonPackage rec {
+    pname = "oil-pyyajl";
+    version = "unreleased";
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "py-yajl";
+      rev = "eb561e9aea6e88095d66abcc3990f2ee1f5339df";
+      sha256 = "17hcgb7r7cy8r1pwbdh8di0nvykdswlqj73c85k6z8m0filj3hbh";
+      fetchSubmodules = true;
+    };
+    # just for submodule IIRC
+    nativeBuildInputs = [ git ];
+  };
+
+  # resholve's primary dependency is this developer build of the oil shell.
+  oildev = python27.pkgs.buildPythonPackage rec {
+    pname = "oil";
+    version = "undefined";
+
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "oil";
+      rev = "ea80cdad7ae1152a25bd2a30b87fe3c2ad32394a";
+      sha256 = "0pxn0f8qbdman4gppx93zwml7s5byqfw560n079v68qjgzh2brq2";
+
+      /*
+      This feels a little dumb; I assume it's wrongish. I was hoping
+      to be able to filter the source down to knock out parts of the
+      source that I'm not using, but I get errors just trying to get
+      *anything* working with the approaches commented out below.
+
+      On IRC eadwu[m] suggested dropping them in extraPostFetch.
+
+      It's not critical to drop most of these; the primary target is
+      the vendored fork of Python-2.7.13, which is ~ 55M and over 3200
+      files, dozens of which get interpreter script patches in fixup.
+      */
+      extraPostFetch = ''
+        #find . -maxdepth 1 -type d | sort
+        rm -rf Python-2.7.13 benchmarks metrics py-yajl rfc gold web testdata services demo devtools cpp
+        # find . -maxdepth 1 -type d | sort
+      ''; # breakers: doc, pgen2, vendor
+    };
+
+    # src = stdenv.lib.cleanSourceWith {
+    #   src = fetchFromGitHub {
+    #     owner = "oilshell";
+    #     repo = "oil";
+    #     rev = "ea80cdad7ae1152a25bd2a30b87fe3c2ad32394a";
+    #     sha256 = "0pxn0f8qbdman4gppx93zwml7s5byqfw560n079v68qjgzh2brq2";
+    #   };
+    #   filter = stdenv.lib.cleanSourceFilter;
+    # };
+
+    # src = stdenv.lib.sourceByRegex (fetchFromGitHub {
+    #   owner = "oilshell";
+    #   repo = "oil";
+    #   rev = "ea80cdad7ae1152a25bd2a30b87fe3c2ad32394a";
+    #   sha256 = "0pxn0f8qbdman4gppx93zwml7s5byqfw560n079v68qjgzh2brq2";
+    # }) [".*"];
+
+    # TODO: not sure why I'm having to set this for nix-build...
+    #       can anyone tell if I'm doing something wrong?
+    SOURCE_DATE_EPOCH = 315532800;
+
+
+    # These aren't, strictly speaking, nix/nixpkgs specific, but I've
+    # had hell upstreaming them. Pulling from resholve source and
+    # passing in from resholve.nix
+    patches = oilPatches;
+
+    buildInputs = [ readline cmark py-yajl makeWrapper ];
+
+    nativeBuildInputs = [ re2c file ];
+
+    # runtime deps
+    propagatedBuildInputs = with python27.pkgs; [ python27 six typing ];
```

Comment:
@thiagokokada Does the comment above at https://github.com/NixOS/nixpkgs/pull/85827#discussion_r550364088 apply here as well?

Reply:
AFAIK, considering the complexity of this package you probably want to inject three parameters in the function input:

- `buildPythonApplication`
- `buildPythonPackage`
- `python27Packages`

And use them accordingly (in this case you would use `with python27Packages; [ six typing ]`. Also, I am not sure why you need to propagate `python27` here.

Reply:
I may be wrong though, maybe someone with more Python package experience can help here.

Reply:
I think you're right on not propagating. Tests are passing local without it, at least. I'll give it a try in CI.

Reply:
I took a swing at this in b0d38949a05; does that look right, or did you intend for me to pass the buildPython functions in from all-packages? (I guess buildPython* are only passed by default for python-modules?)

Reply:
Actually, this way looks mostly good. I just don't know what is better for those packages that will be stuck with Python 2.7 in future, if it is to use `pythonPackages` or `python27Packages`. I think the later is preferred because eventually we will switch to `pythonPackages` =  `python3Packages`, but I may be wrong.

Reply:
Of course, you can also just leave as is and do `pythonPackages = python27Packages` in your `all-packages.nix` declaration.

Reply:
I would just use ``python27Packages`` as an input because you can't overwrite it right now anyway.

Reply:
@SuperSandro2000 Do you know if/how that should affect `disabled = !python27Packages.isPy27;` (ex: https://github.com/NixOS/nixpkgs/pull/85827/files#diff-f19dcdf25b44e15017ea999a1f918bb8f0485bf41e700a6b557e5b7c5c319669R41)?

Reply:
I updated these to python27Packages in 5bbea01. I'm not sure about `disabled = !python27Packages.isPy27;`, but it worked, so I'll tentatively resolve this.

Reply:
Yeah, not problem in maintaining the `disabled = !python27Packages.isPy27`, makes it clear that the package is never meant to be built with Python 3.